subsys/bluetooth/audio/ascs.c

```diff
@@ -45,62 +46,27 @@
 	(_id > CONFIG_BT_ASCS_ASE_SNK_COUNT ? BT_UUID_ASCS_ASE_SRC : BT_UUID_ASCS_ASE_SNK)
 #define ASE_COUNT (CONFIG_BT_ASCS_ASE_SNK_COUNT + CONFIG_BT_ASCS_ASE_SRC_COUNT)
 
+BUILD_ASSERT(ASE_COUNT <= CONFIG_BT_ISO_MAX_CHAN,
+	     "ASCS total ASE count exceeds ISO BT_ISO_MAX_CHAN");
```

Comment:
But not every ASE has to be in <= QoS state right? So the check will still be invalid then.

Reply:
Not sure I follow your comment. 

I think it makes sense to restrict the total number of ASEs to the total number of `audio_iso`s.

Reply:
Hmm, I'm starting to think that this assert is correct. If the server exposes e.g. 4 ASE it means that it can handle => 4 CIS connections. However, if the number of ACL connections is e.g. 3, the number of CIS and non-idle ASE would be still 4. It's because even if there are more clients connected the number of audio endpoints that can be in use will be the same.

Reply:
Keep in mind that `ASE_COUNT` is the number of endpoints _per_ ACL connection. 

So if `ASE_COUNT=4` and `CONFIG_BT_MAX_CONN=3`, then we have a total of 12 ASEs